src/charm.py

```diff
@@ -353,6 +383,66 @@ def _config_file_content_matches(self, content: str) -> bool:
         existing_content = self._container.pull(path=f"{BASE_CONFIG_PATH}/{CONFIG_FILE_NAME}")
         return existing_content.read() == content
 
+    def _on_get_profile_a_home_network_public_key_action(self, event: ActionEvent) -> None:
+        if not self._container.can_connect():
+            event.fail(message="Container is not ready yet.")
+            return
+        if not self._profile_a_home_network_private_key_stored():
+            event.fail(message="Home network private key is not stored yet.")
+            return
+        event.set_results(
+            {
+                "public-key": self._get_profile_a_home_network_public_key(),  # type: ignore[arg-type] # noqa: E501
+            }
+        )
+
+    def _generate_profile_a_home_network_private_key(self) -> None:
```

Comment:
```suggestion
    def _generate_home_network_private_key(self) -> None:
```

Reply:
The profile A/B thing is not even implemented in SD-Core so I don't think it's worth specifying it

Reply:
This also applies elsewhere in this file